torch/csrc/autograd/cpp_hook.cpp

```diff
@@ -0,0 +1,49 @@
+#include <torch/csrc/autograd/cpp_hook.h>
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/autograd/custom_function.h>
+
+using torch::autograd::Variable;
+static void check_single_result (Variable value, Variable result, std::string hook_name);
```

Comment:
FYI it's more idiomatic in C++ to put static functions in an anonymous namespace